docs/src/components/BuiltinSearch/index.js

```diff
@@ -0,0 +1,162 @@
+
+  const allRows = useMemo(() => {
+    return (Object.keys(builtins._categories)
+      .map((category) => {
+        return builtins._categories[category].map((builtin) => {
+          const fn = builtins[builtin];
+          if (!fn) return null;
+
+          // we have links out there in the wild that use the name without a dot
+          // this needs to be preserved for backwards compatibility.
+          const anchor = `builtin-${category}-${builtin.replaceAll(".", "")}`;
+
+          const isInfix = !!fn.infix;
+          const isRelation = !!fn.relation;
+
+          const args = fn.args || [];
+          const result = fn.result || {};
+
+          const signature = isInfix
+            ? `${args[0]?.name || "x"} ${fn.infix} ${args[1]?.name || "y"}`
+            : isRelation
+            ? `${builtin}(${args.map((a) => a.name).join(", ")}, ${result.name})`
+            : `${result.name || "result"} := ${builtin}(${args.map((a) => a.name).join(", ")})`;
+          return {
+            name: builtin,
+            category: category,
+            wasm: fn.wasm,
+            introduced: fn.introduced,
+            infix: isInfix,
+            signature: signature,
+            anchor: anchor,
+            versions: fn.available,
+          };
+        });
+      })).reduce((a, b) => a.concat(b), []);
+  });
+  const filteredRows = allRows.map((row) => {
+    return ((row.name.includes(searchTerm) || row.signature.includes(searchTerm) && row.infix)
+        && (!selectedCategory || row.category == selectedCategory) && (!isWasm || row.wasm)
+        && (row.versions.includes(selectedVersion)))
+      ? row
+      : null;
+  }).filter((a) => a != null && (searchTerm || selectedCategory || isWasm || selectedVersion));
+  return (
+    <>
+      <div className={styles.searchBar}>
+        <select
+          title="Filter Catgeory"
+          value={selectedCategory}
+          onChange={(e) => setSelectedCategory(e.target.value)}
+        >
+          <option value="" key="0">All Categories</option>
+          {Array.from(Object.keys(builtins._categories).entries()).map((mod) => (
+            <option key={mod[0] + 1} value={mod[1]}>{capitalise(mod[1])}</option>
+          ))}
+        </select>
+        <select
+          title="Filter Version"
+          value={selectedVersion}
+          onChange={(e) => setSelectedVersion(e.target.value)}
+        >
+          <option key="0" value="edge">Edge</option>
+          {Array.from(versions.toReversed().entries()).map((ver) => (
+            <option key={ver[0] + 1} value={ver[1]}>{ver[1]}</option>
+          ))}
+        </select>
+        <input
+          type="text"
+          title="Function Name"
+          placeholder="search builtins"
+          value={searchTerm}
+          onChange={(e) => setSearchTerm(e.target.value)}
+        />
+        <span className={isWasm ? styles.wasm_1 : styles.wasm_0} onClick={(e) => setIsWasm(!isWasm)}>
+          Wasm
+        </span>
```

Comment:
I think I would like to see this as a checkbox. At the moment it's not clear immediately if it's a button.

Reply:
I couldn't really figure out how to communicate that it was wasm related clearly without making it feel even more cluttered, maybe adding the checkbox or a check/cross symbol in there would make it more intuitive